documentation/quarkus/guide-tkit.asciidoc

```diff
@@ -0,0 +1,192 @@
+You can enable or disable the formatting by setting `quarkus.tkit.log.console.json` in the properties.
+
+[source, xml]
+----
+<dependency>
+  <groupId>org.tkit.quarkus</groupId>
+  <artifactId>tkit-quarkus-log-json</artifactId>
+  <version>1.5.0</version>
+</dependency>
+----
+
+== tkit-quarkus-jpa
+
+The `tkit-quarkus-jpa` extension provides base classes for JPA entities (`org.tkit.quarkus.jpa.models.TraceableEntity`) and link:../guide-dao.asciidoc[DAOs] (`org.tkit.quarkus.jpa.daos.AbstractDAO<T>`).
+
+The `TraceableEntity` implements a string based `ID` and traceable fields such as `version`, `creationDate`, `creationUser` or `modificationDate`. +
+CRUD operation following the DAO pattern are provided by `AbstractDAO<T>`. Your DAO class can simply extend the base class and only needs to implement the business logic.
+
+[source, xml]
+----
+<dependency>
+  <groupId>org.tkit.quarkus</groupId>
+  <artifactId>tkit-quarkus-jpa</artifactId>
+  <version>2.8.0</version>
+</dependency>
+----
+
+== tkit-quarkus-rest
+
+This extension provides link:../guide-dto.asciidoc[DTOs] for entities (`org.tkit.quarkus.rs.models.TraceableDTO`), exceptions (`org.tkit.quarkus.rs.models.TraceableDTO.RestExceptionDTO`) and paged results (`tkit.quarkus.rs.models.TraceableDTO.PageResultDTO<T>`).
+
+Furthermore, it provides an `DefaultExceptionMapper` to log the rest exceptions.
+
+[source, xml]
+----
+<dependency>
+  <groupId>org.tkit.quarkus</groupId>
+  <artifactId>tkit-quarkus-rest</artifactId>
+  <version>2.4.0</version>
+</dependency>
+----
+
+== tkit-quarkus-test
+
+The `tkit-quarkus-test` provides functionality to simplify test setup based on link:https://www.testcontainers.org/[Testcontainers].
+
+[source, xml]
+----
+<dependency>
+  <groupId>org.tkit.quarkus</groupId>
+  <artifactId>tkit-quarkus-test</artifactId>
+  <version>1.13.0</version>
+</dependency>
+----
+
+You can specify which services you want to start before running the test in a docker-compose.yaml file under `src/main/resources.`
+The following example can be used to start a Postgresql database before testing.
+
+[source,yaml]
+----
+version: '3.9'
+services:
+  demo-db:
+    container_name: demo-db
+    image: postgres:11.5
+    environment:
+      POSTGRES_DB: "demo"
+      POSTGRES_USER: "demo"
+      POSTGRES_PASSWORD: "demo"
+    labels:
+      - "test.priority=90"
+      - "test.Wait.forLogMessage.regex=.*database system is ready to accept connections.*\\s"
+      - "test.Wait.forLogMessage.times=2"
+      - "test.log=true"
+      - "test.property.quarkus.datasource.jdbc.url=jdbc:postgresql://$${host:demo-db}:$${port:demo-db:5432}/demo?sslmode=disable"
+----
+
+Create an abstract base class for your tests that takes care of setting up the environment:
+
+[source, Java]
+----
+@QuarkusTestcontainers
+@QuarkusTestResource(DockerComposeTestResource.class)
+public abstract class AbstractTest {
+
+    @DockerService("tkit-parameter")
```

Comment:
this is only required in native mode testing -when we test our application as a docker image i.e. full integration test. For ,,regular" jvm tests, it is sufficient to annotate a test class with `@QuarkusTestResource(DockerComposeTestResource.class)`